modules/metastore/tests/src/Kernel/Controller/MetastoreAccessManagerTest.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+namespace Drupal\Tests\metastore\Controller\Kernel;
+
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\metastore\Controller\MetastoreAccessManager;
+use Drupal\metastore\NodeWrapper\Data;
+use Drupal\metastore\NodeWrapper\NodeDataFactory;
+use Drupal\node\NodeInterface;
+use Drupal\Tests\user\Traits\UserCreationTrait;
+use MockChain\Chain;
+use MockChain\Options;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Tests the MetastoreAccessManager.
+ *
+ * Note, no current test for the canViewRevision() method, because it relies
+ * on static methods from the Data wrapper, which can't be mocked.
+ *
+ * @coversDefaultClass \Drupal\metastore\Controller\MetastoreAccessManager
+ *
+ * @group dkan
+ * @group metastore
+ */
+class MetastoreAccessManagerTest extends KernelTestBase {
+
+  use UserCreationTrait;
+
+  /**
+   * The MetastoreAccessManager service.
+   */
+  protected MetastoreAccessManager $accessManager;
+
+  /**
+   * A user with permissions to work with the metastore.
+   */
+  protected AccountInterface $priviledgedUser;
+
+  /**
+   * A user without permissions to work with the metastore.
+   */
+  protected AccountInterface $unpriviledgedUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'metastore',
+    'common',
+    'dkan',
+    'field',
+    'node',
+    'user',
+    'workflows',
+    'content_moderation',
+    'system',
+    'text',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp(): void {
+    parent::setUp();
+    // Set up the necessary services and configurations for the test.
+    $this->installConfig(['node', 'metastore']);
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('user');
+
+    // Mock the NodeDataFactory to return a Data wrapper with a data node mock.
+    $itemFactory = (new Chain($this))
+      ->add(NodeDataFactory::class, 'getInstance', (new Options)
+        ->add('123', Data::class)
+        ->add('345', NULL)
+        ->index(0)
+      )
+      ->add(Data::class, 'fix', NULL)
+      ->add(Data::class, 'getEntity', NodeInterface::class)
+      ->add(NodeInterface::class, 'bundle', 'data')
+      ->add(NodeInterface::class, 'get', FieldItemListInterface::class)
+      ->add(NodeInterface::class, 'language', LanguageInterface::class)
+      ->add(NodeInterface::class, 'getEntityType', EntityTypeInterface::class)
+      ->add(EntityTypeInterface::class, 'id', 'node')
+      ->add(LanguageInterface::class, 'getId', 'en')
+      ->add(FieldItemListInterface::class, 'getString', 'dataset')
+      ->getMock();
+
+    $this->container->set('dkan.metastore.metastore_item_factory', $itemFactory);
+
+    $this->priviledgedUser = $this->createUser([
```

Comment:
For each individual test, I'd create a user that has the minimum permissions per test. So `testCanCreate` would create a user with only the permissions required to create, and then the other test methods would create a user based on the minim required for their access check.